lib/ain-grpc/src/rpc.rs

```diff
@@ -216,6 +220,59 @@ impl MetachainRPCServer for MetachainRPCModule {
             })
     }
 
+    fn get_pending_transaction(&self) -> RpcResult<Vec<EthPendingTransactionInfo>> {
```

Comment:
Can we keep the conversion in the `impls.rs` file so we can keep `rpc.rs` clean 

```
    fn get_pending_transaction(&self) -> RpcResult<Vec<EthPendingTransactionInfo>> {
        get_pool_transactions()
            .map(|txs| {
                txs.into_iter()
                    .flat_map(|tx| EthPendingTransactionInfo::try_from(tx.as_str()))
                    .collect()
            })
            .map_err(|e| Error::Custom(e.to_string()))
    }
 ```
 
 in impls.rs :
 
 ```
 
impl TryFrom<&str> for EthPendingTransactionInfo {
    type Error = TransactionError;

    fn try_from(raw_tx: &str) -> Result<Self, Self::Error> {
        let signed_tx: SignedTx = raw_tx.try_into()?;

        let to = if let Some(signed_to) = signed_tx.to() {
            String::from("0x") + &hex::encode(signed_to.as_fixed_bytes())
        } else {
            String::from("null")
        };

        let input = if signed_tx.data().len() > 0 {
            format!("0x{}", hex::encode(signed_tx.data()))
        } else {
            String::from("0x0")
        };

        let pending_transaction = EthPendingTransactionInfo {
            hash: format!(
                "0x{}",
                hex::encode(signed_tx.transaction.hash().as_fixed_bytes())
            ),
            nonce: format!("0x{}", signed_tx.nonce().to_string()),
            block_hash: String::from(
                "0000000000000000000000000000000000000000000000000000000000000000",
            ),
            block_number: String::from("null"),
            transaction_index: String::from("0x0"),
            from: format!("0x{}", hex::encode(signed_tx.sender.as_fixed_bytes())),
            to,
            value: format!("0x{}", signed_tx.value().to_string()),
            gas: format!("0x{}", signed_tx.gas_limit().to_string()),
            gas_price: format!("0x{}", signed_tx.gas_price().to_string()),
            input,
            v: format!("0x{:x}", signed_tx.v()),
            r: format!("0x{}", hex::encode(signed_tx.r().as_fixed_bytes())),
            s: format!("0x{}", hex::encode(signed_tx.s().as_fixed_bytes())),
        };

        Ok(pending_transaction)
    }
}
```

Reply:
Top tip. Thanks Jeremy.